src/presets/src/Presets.cpp

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) 2022 FlyByWire Simulations
+// SPDX-License-Identifier: GPL-3.0
+
+#include "Presets.h"
+#include "Aircraft/AircraftPreset.h"
+#include "Lighting/LightPreset.h"
+
+Presets PRESETS;
+
+/**
+ * Gauge Callback
+ * @see
+ * https://docs.flightsimulator.com/html/Content_Configuration/SimObjects/Aircraft_SimO/Instruments/C_C++_Gauges.htm?rhhlterm=_gauge_callback&rhsearch=_gauge_callback
+ */
+__attribute__((export_name("Presets_gauge_callback"))) extern "C" __attribute__((unused)) bool
+Presets_gauge_callback(__attribute__((unused)) FsContext ctx, int service_id, void* pData) {
+  switch (service_id) {
+    case PANEL_SERVICE_PRE_INSTALL: {
+      return true;
+    }
+    case PANEL_SERVICE_POST_INSTALL: {
+      return PRESETS.initialize();
+    }
+    case PANEL_SERVICE_PRE_DRAW: {
+      auto drawData = static_cast<sGaugeDrawData*>(pData);
+      return PRESETS.onUpdate(drawData->dt);
+    }
+    case PANEL_SERVICE_PRE_KILL: {
+      return PRESETS.shutdown();;
+    }
+    default:
+      break;
+  }
+  return false;
+}
+
+bool Presets::initialize() {
+  std::cout << "PRESETS: Connecting to SimConnect..." << std::endl;
+
+  lightPresetPtr = std::make_unique<LightPreset>();
+  aircraftPresetPtr = std::make_unique<AircraftPreset>();
+
+  if (!SUCCEEDED(SimConnect_Open(&hSimConnect, "Presets", nullptr, 0, 0, 0))) {
+    std::cout << "PRESETS: SimConnect failed." << std::endl;
+    return false;
+  }
+
+  isConnected = true;
+
+  // Simulation Data to local data structure mapping
+  const HRESULT result = SimConnect_AddToDataDefinition(hSimConnect, DataTypesID::SimulationDataTypeId, "SIMULATION TIME", "NUMBER");
+
+  // initialize modules
+  lightPresetPtr->initialize();
+  aircraftPresetPtr->initialize();
+
+  std::cout << "PRESETS: SimConnect connected." << std::endl;
+  return (result == S_OK);
+}
+
+bool Presets::onUpdate(double deltaTime) {
+  if (isConnected) {
+
+    // read simulation data from simconnect
+    simConnectRequestData();
+    simConnectProcessMessages();
+
+    // detect pause
+    if (simulationData.simulationTime == previousSimulationTime || simulationData.simulationTime < 0.2) {
+      // pause detected -> return
+      return true;
+    }
+    // store previous simulation time
```

Comment:
For example this comment has no value for me, since it's obvious by reading the code itself.

Reply:
I agree with this one :) copy&pasted this one
I like the WHAT and WHY guidance - thanks